### PR TITLE
JSM: Add modules and TS files for LightningStrike, LightningStorm

### PR DESCRIPTION
**Pull request by yomboprime**
I fixed a small bug in the HTML example where variable `rayLength` was not declared.

TS files are not tested.

### CODE REVIEW

File: examples/jsm/geometries/LightningStrike.d.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  constructor(rayParameters?: RayParameters);
+  copyParameters( dest?: RayParameters, source?: RayParameters): RayParameters;
+
+  // Ray states
+  static readonly RAY_INITIALIZED: number;
+  static readonly RAY_UNBORN: number;
+  static readonly RAY_PROPAGATING: number;
+  static readonly RAY_STEADY: number;
+  static readonly RAY_VANISHING: number;
+  static readonly RAY_EXTINGUISHED: number;
```
Comment (yomboprime) [inline examples/jsm/geometries/LightningStrike.d.ts]: Is it okay to use `static readonly` for class constants?

Comment (Mugen87) [reply examples/jsm/geometries/LightningStrike.d.ts]: I've not used `readonly` so far but I think it makes sense in this context.

File: examples/jsm/geometries/LightningStrike.d.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  roughness: number;
+  straightness: number;
+  propagationTimeFactor: number;
+  vanishingTimeFactor: number;
+  endPropagationTime: number;
+  beginVanishingTime: number;
+}
+
+export interface RayParameters {
+  sourceOffset: Vector3 | undefined;
```
Comment (Mugen87) [inline examples/jsm/geometries/LightningStrike.d.ts]: If a parameter is optional it's better to write it like so: `sourceOffset?: Vector3`

File: examples/jsm/geometries/LightningStrike.d.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  static readonly RAY_STEADY: number;
+  static readonly RAY_VANISHING: number;
+  static readonly RAY_EXTINGUISHED: number;
+
+  state: number;
+
+  update(time: number): void;
+
+  copy(source: LightningStrike): LightningStrike;
+  clone(void): LightningStrike;
```
Comment (Mugen87) [inline examples/jsm/geometries/LightningStrike.d.ts]: If a method has no parameter, it's  better to declare it like so: `clone(): LightningStrike;`. `void` is not needed.

File: examples/jsm/objects/LightningStorm.d.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  maxSlope: number | undefined;
+
+  maxLightnings: number | undefined;
+
+  lightningMinPeriod: number | undefined;
+  lightningMaxPeriod: number | undefined;
+  lightningMinDuration: number | undefined;
+  lightningMaxDuration: number | undefined;
+
+  lightningParameters: RayParameters | undefined;
```
Comment (Mugen87) [inline examples/jsm/objects/LightningStorm.d.ts]: `RayParameters` is undefined.

File: examples/jsm/objects/LightningStorm.d.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  lightningMaterial: Material | undefined;
+
+  isEternal: boolean | undefined;
+
+  onRayPosition?: (source: Vector3, dest: Vector3) => void;
+  onLightningDown?: (lightning: LightningStrike) => void;
```
Comment (Mugen87) [inline examples/jsm/objects/LightningStorm.d.ts]: `LightningStrike` is also undefined. You have to import the type like so:

```js
import { LightningStrike } from '../geometries/LightningStrike';
```

File: examples/jsm/geometries/LightningStrike.d.ts (hunk 2 above)
Comment (yomboprime) [reply examples/jsm/geometries/LightningStrike.d.ts]: Done.

File: examples/jsm/geometries/LightningStrike.d.ts (hunk 3 above)
Comment (yomboprime) [reply examples/jsm/geometries/LightningStrike.d.ts]: Done.

File: examples/jsm/objects/LightningStorm.d.ts (hunk 4 above)
Comment (yomboprime) [reply examples/jsm/objects/LightningStorm.d.ts]: Done.

File: examples/jsm/objects/LightningStorm.d.ts (hunk 5 above)
Comment (yomboprime) [reply examples/jsm/objects/LightningStorm.d.ts]: Done. Thanks!

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**